cmd/podman/compose.go

```diff
@@ -132,71 +124,23 @@ func composeDockerHost() (string, error) {
 		}
 	}
 
-	parsedConnection, err := url.Parse(connection.URI)
+	parsedConnection, err := url.Parse(conf.URI)
 	if err != nil {
 		return "", fmt.Errorf("preparing connection to remote machine: %w", err)
 	}
 
 	// If the default connection does not point to a `podman
 	// machine`, we cannot use a local path and need to use SSH.
-	if !connection.IsMachine {
-		// Compose doesn't like paths, so we optimistically
+	if !conf.MachineMode {
+		// Docker Compose v1 doesn't like paths for ssh, so we optimistically
 		// assume the presence of a Docker socket on the remote
 		// machine which is the case for podman machines.
-		return strings.TrimSuffix(connection.URI, parsedConnection.Path), nil
-	}
-
-	machineProvider, err := provider.Get()
-	if err != nil {
-		return "", fmt.Errorf("getting machine provider: %w", err)
-	}
-	dirs, err := machine.GetMachineDirs(machineProvider.VMType())
-	if err != nil {
-		return "", err
-	}
-
-	machineList, err := vmconfigs.LoadMachinesInDir(dirs)
-	if err != nil {
-		return "", fmt.Errorf("listing machines: %w", err)
-	}
-
-	// Now we know that the connection points to a machine and we
-	// can find the machine by looking for the one with the
-	// matching port.
-	connectionPort, err := strconv.Atoi(parsedConnection.Port())
-	if err != nil {
-		return "", fmt.Errorf("parsing connection port: %w", err)
-	}
-	for _, item := range machineList {
-		if connectionPort != item.SSH.Port {
-			continue
-		}
-
-		state, err := machineProvider.State(item, false)
-		if err != nil {
-			return "", err
-		}
-
-		if state != define.Running {
-			return "", fmt.Errorf("machine %s is not running but in state %s", item.Name, state)
+		if parsedConnection.Scheme == "ssh" {
+			return strings.TrimSuffix(conf.URI, parsedConnection.Path), nil
 		}
-
-		// TODO This needs to be wired back in when all providers are complete
-		// TODO Need someoone to plumb in the connection information below
-		// if machineProvider.VMType() == define.WSLVirt || machineProvider.VMType() == define.HyperVVirt {
-		// 	if info.ConnectionInfo.PodmanPipe == nil {
-		// 		return "", errors.New("pipe of machine is not set")
-		// 	}
-		// 	return strings.Replace(info.ConnectionInfo.PodmanPipe.Path, `\\.\pipe\`, "npipe:////./pipe/", 1), nil
-		// }
-		// if info.ConnectionInfo.PodmanSocket == nil {
-		// 	return "", errors.New("socket of machine is not set")
-		// }
-		// return "unix://" + info.ConnectionInfo.PodmanSocket.Path, nil
-		return "", nil
+		return conf.URI, nil
 	}
-
-	return "", fmt.Errorf("could not find a matching machine for connection %q", connection.URI)
+	return getMachineConn(conf.URI, parsedConnection)
```

Comment:
Maybe wrap the error here with something a bit more helpful to users? "could not get connection to a podman machine: machine myMachine is not running but is in state stopped" is a lot more friendly than just "machine myMachine is not running but is in state stopped"

Reply:
yeah, will add this if I have to repush.
The whole get the machine socket seems rather iffy to me. No idea why we loop over all machines and match the port at all.

Quite frankly the longer I look at this I would just call docker-compose v1 unsupported and juts use the ssh connection URI as this works find with composev2. This would allow us to remove all the machine special handling.

Reply:
I'd prefer to have a deprecation period for that. Maybe formally deprecate in 5.1, remove in 5.2?

Reply:
yeah not looking at removing this right now but I am going to keep saying v1 is deprecated upstream for a long time and not supported so we shouldn't bother IMO